Support to compile from source as string and testcases to test the same

arjunballa opened the pull request:
Support to compile from source as string and testcases to test the same

arjunballa committed:
Support to compile from source as string

arjunballa committed:
Test cases for compiling from raw string

arjunballa committed:
- added documentation
- refactored log file

cpopov commented:
Hello, can you explain what you are trying to achieve? Also there is problem with diff, I can'r see the differences on github, I bet it is due to unix/windows differences in new lines.

arjunballa commented:
I have added support to compile less from string source.
In my project we store the less source and imports in database as a blob and we need to dynamically compile less and serve.
To do this I modified to LessSource.java where all imports should be provided to public LessSource(String source, Map<String, LessSource> imports) constructor and also changed resolveImports method to look for imports in map.

I will submit a new pull request so that you can compare.

arjunballa commented:
Can you please export the eclipse Java Formatter profile and push that to github. I am using eclipse default format and there are lot of formatting differences.
My eclipse version is Helios Service Release 2

cpopov commented:
I use intellij and Eclipe. You don't need the settings file. Don;t reformat the whole code, just your piece. Also I think your number of columns is 80. You can just increase that 120 or more.

arjunballa commented:
As the design is changed by introducing a new abstract class Resource this pull request is not valid anymore.
This pull request can be closed.
